### PR TITLE
Load modules relative to the app base path

### PR DESCRIPTION
**Pull request by Pchelolo**
Currently in module/spec loading we rely on the fact that `router` is located in `/lib/router.js`. However, when we move lib directory to the framework, that will not be the case any more. To solve the problem we will get an appBasePath from service-runner and resolve everything from it.

Tests will not pass until https://github.com/wikimedia/service-runner/pull/80 is merged and published.

### CODE REVIEW

File: test/features/router/buildTree.js [hunk 1]
```diff
@@ -20,71 +20,20 @@ var rootSpec = {
     }
 };
 
-// x-subspec and x-subspecs is no longer supported.
-var faultySpec = {
```
Comment (Pchelolo) [inline test/features/router/buildTree.js]: Not really related, just a small cleanup

**Comment by Pchelolo** [comment]
cc @wikimedia/services

**Comment by d00rman** [comment]
Hm, this will not work in prod, because `basePath` there is one level up from the repo's location in the directory tree. The way to get around this is to tell `service-runner` what `basePath` really is, as it depends on the environment it is run in.

File: lib/router.js [hunk 2]
```diff
@@ -558,35 +542,27 @@ Router.prototype.handleResources = function(hyper) {
...
+ * @param {Object} spec specification object
+ * @param {HyperSwitch} hyper HyperSwitch context
+ * @returns {Promise.<Router>} a router with loaded spec
  */
 Router.prototype.loadSpec = function(spec, hyper) {
     var self = this;
     var rootNode = new Node();
-    var specPromise;
-    if (spec && spec.constructor === String) {
-        specPromise = this._readSpec(spec);
```
Comment (Pchelolo) [inline lib/router.js]: This doesn't happen in reality, so it's dead code and can be removed.

**Comment by Pchelolo** [comment]
This would require a config change on next deploy, because now we try to load all modules from the base_path, which in production is in the deploy repo root. So, we need to add an `app_base_path` property to the config.

**Comment by d00rman** [comment]
@Pchelolo I think this deserves a version bump to articulate a slight incompatibility with 0.9.x. Let's go for 0.10.0 ?

**Comment by Pchelolo** [comment]
@d00rman done

**Comment by d00rman** [comment]
@eevans @gwicke any objections against merging this? Mental note: we'll need config changes for this.

**Pull request merged**